specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/disk.json

```diff
@@ -0,0 +1,2908 @@
+        },
+        "x-ms-examples": {
+          "List all disk access resources in a resource group.": {
+            "$ref": "./examples/ListDiskAccessesInAResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Compute/diskAccesses": {
+      "get": {
+        "tags": [
+          "DiskAccesses"
+        ],
+        "operationId": "DiskAccesses_List",
+        "description": "Lists all the disk access resources under a subscription.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/DiskAccessList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all disk access resources in a subscription.": {
+            "$ref": "./examples/ListDiskAccessesInASubscription.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/diskAccesses/{diskAccessName}/privateLinkResources": {
+      "get": {
+        "tags": [
+          "DiskAccesses"
+        ],
+        "operationId": "DiskAccesses_GetPrivateLinkResources",
+        "description": "Gets the private link resources possible under disk access resource",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/DiskAccessNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/PrivateLinkResourceListResult"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List all possible private link resources under disk access resource.": {
+            "$ref": "./examples/GetDiskAccessPrivateLinkResources.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/restorePointCollections/{restorePointCollectionName}/restorePoints/{vmRestorePointName}/diskRestorePoints/{diskRestorePointName}": {
+      "get": {
+        "tags": [
+          "IncrementalRestorePoints"
```

Comment:
how/when do these get created?

Reply:
They are created by CRP by calling DiskRP's AllocateRestorePoint API.

CRP passes restorePointCollectionName and vmRestorePointName along with a list of disks of the VM that need to take restore point to DiskRP. Then DiskRP creates snapshot of these disks, stores them as restore point objects and returns armId back to CRP.

The newly added GET disk restore point API exposes disk restore point resource directly to customer because we want to allow customer to directly retrieve SAS of disk restore point. That functionality is currently delegated to CRP.